app/src/main/java/io/gnosis/safe/utils/ParamSerializer.kt

```diff
@@ -0,0 +1,35 @@
+package io.gnosis.safe.utils
+
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.Types
+import io.gnosis.data.backend.dto.DataDecodedDto
+import io.gnosis.data.backend.dto.ValueDecodedDto
+
+class ParamSerializer(
+    moshi: Moshi
+) {
+    private val decodedDataAdapter: JsonAdapter<DataDecodedDto>
+    private val decodedValuesAdapter: JsonAdapter<List<ValueDecodedDto>>
+
+    init {
+        decodedDataAdapter = moshi.adapter(DataDecodedDto::class.java)
+        decodedValuesAdapter = moshi.adapter(Types.newParameterizedType(List::class.java, ValueDecodedDto::class.java))
+    }
+
+    fun serializeDecodedData(decodedData: DataDecodedDto): String {
+        return decodedDataAdapter.toJson(decodedData)
+    }
+
+    fun unserializeDecodedData(decodedDataString: String): DataDecodedDto? {
```

Comment:
I would call it `deserializeDecodedData()` and `deserializeDecodedValues()` but that's not a blocker.